components/file_preview.jsx

```diff
@@ -114,26 +111,80 @@ export default class FilePreview extends React.PureComponent {
         });
 
         this.props.uploadsInProgress.forEach((clientId) => {
+            let percent = 0;
+            let fileNameComponent;
+            let previewImage;
+            const uploadsInProgress = this.props.uploadsProgressPercent;
+            const fileInfo = uploadsInProgress[clientId];
+            if (fileInfo) {
+                percent = fileInfo.percent;
+                const percentTxt = ` (${percent.toFixed(0)}%)`;
+                const fileType = getFileTypeFromMime(fileInfo.extension);
+                previewImage = <div className={'file-icon ' + Utils.getIconClassName(fileType)}/>;
+
+                fileNameComponent = (
```

Comment:
Could this be refactored out into its own component? That might make it easier to test on its own